packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -2684,13 +2655,25 @@ class _PaddleSliderTrackShapePathPainter {
     final Offset topLeft = (topLobeRect.topLeft * scale) + center;
     final Offset bottomRight = (topLobeRect.bottomRight * scale) + center;
     double shift = 0.0;
-    if (topLeft.dx < edgeMargin) {
-      shift = edgeMargin - topLeft.dx;
+
+    final double startGlobal = parentBox.localToGlobal(Offset.zero).dx;
+    if (topLeft.dx < startGlobal + edgeMargin) {
+      shift = startGlobal + edgeMargin - topLeft.dx;
     }
-    if (bottomRight.dx > parentBox.size.width - edgeMargin) {
-      shift = parentBox.size.width - bottomRight.dx - edgeMargin;
+
+    final double endGlobal = parentBox.localToGlobal(Offset(parentBox.size.width, parentBox.size.height)).dx;
+    if (bottomRight.dx > endGlobal - edgeMargin) {
+      shift = endGlobal - edgeMargin - bottomRight.dx;
     }
+
     shift = scale == 0.0 ? 0.0 : shift / scale;
+    if (shift < 0.0) {
+      // shifting to the left
```

Comment:
Capital letter and period here and below :apple: :pencil2: :100:

Reply:
Done